src/shared/components/dateRangePicker/DateRangePicker.tsx

```diff
@@ -108,11 +124,27 @@ class DateRangePicker extends PureComponent<Props, State> {
   }
 
   private handleSelectLower = (lower: string): void => {
-    this.setState({lower})
+    this.setState(prev => ({
+      ...prev,
+      lower,
+      validDateRange: this.isTimeRangeValid(lower, this.state.upper),
+    }))
   }
 
   private handleSelectUpper = (upper: string): void => {
-    this.setState({upper})
+    this.setState(prev => ({
+      ...prev,
+      upper,
+      validDateRange: this.isTimeRangeValid(this.state.lower, upper),
+    }))
+  }
+
+  private isTimeRangeValid = (lower: string, upper: string): boolean => {
+    return moment(lower).isBefore(upper)
```

Comment:
Really great work on the PR. We can totally avoid using moment here by simply doing something like:

‘’’
new Date(lower).getTime() < new Date(upper).getTime()
‘’’

this effectively converts the date string to a millisecond and compares the two

Reply:
Very true!! I'll add that change 👍